### ISSUE TITLE
Sub-selections on a null field should just be null, not cause errors

### ISSUE THREAD

**Issue by kaqqao** [graphql-java/graphql-java-tools#34]
If a field is resolved to a `null`, and the query wants a sub-selection of it, all the sub-field values should also just be `null`, not cause errors. Yet, if I try such a thing I get an error for each of the subfields.

Here's an example:

```java
public class LinkResolver implements GraphQLResolver<Link> {
    
    private final UserRepository userRepository;

    public LinkResolver(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public User postedBy(Link link) {
        if (link.getUserId() == null) {
            return new User("", "", "", ""); // <----- this is the interesting line
        }
        return userRepository.findById(link.getUserId());
    }
}
```

If that line is replaced by `return null`, the result for a query like:

```
{
    allLinks {
        url
        postedBy {
           name
           email
        }
    }
}
```

would contain an error for each `name` and `email` for a `postedBy` that is `null`.

**Comment by apottere** [comment]
@kaqqao That link is 404ing, do you have a new link?

**Comment by apottere** [comment]
Relevant issue: #26

**Comment by kaqqao** [comment]
@apottere I've just realized that repo is private. I've now forked it and updated the link.

**Comment by apottere** [comment]
I'm not sure how they differ since you didn't post the graphql schema associated with the classes, but I wrote a test case in 154791d94a8acf2881960704c8caefea9ef485b3 that seems to cover this issue and it passes.  Is `User` a nullable or non-null field?

**Comment by kaqqao** [comment]
You're absolutely right. I tested again and can't reproduce it :/
I'm terribly sorry to have wasted your time. And thanks a bunch for the fixes in 3.1.3!

**Comment by apottere** [comment]
No worries, I got a good test case out of it :)
